pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -232,10 +232,11 @@ func (c cacheStats) merge(other cacheStats) cacheStats {
 	}
 }
 
-func buildClusterKey(service *model.Service, port *model.Port, cb *ClusterBuilder) *clusterCache {
+func buildClusterKey(service *model.Service, port *model.Port, cb *ClusterBuilder, efKeys []string) *clusterCache {
 	clusterName := model.BuildSubsetKey(model.TrafficDirectionOutbound, "", service.Hostname, port.Port)
 	clusterKey := &clusterCache{
 		clusterName:     clusterName,
+		proxyVersion:    cb.proxy.Metadata.IstioVersion,
```

Comment:
Why do we need this one?

Reply:
Since I think its already pre-filtered in https://github.com/istio/istio/blob/2f84ae2646764f4bea6a1afcf455c95b824375b3/pilot/pkg/model/push_context.go#L1663?

Reply:
In theory each envoyfilter can include multi patches, and we allow proxy match for each proxy. 

So even though a envoyfilter can be matched by two proxies, but the patches can be different.

Reply:
I am not sure I understand. EnvoyFilterWrapper comes AFTER filtering the patches to only the ones used by the proxy. Then we look at all the patches in EnvoyFilterWrapper and form a key.

So how can they be different?